From 265dc93984c2d6513c0618582d898e3eadc60b0a Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Fri, 15 Aug 2025 23:18:18 +0800
Subject: [PATCH 32/63] loonggpu: Pass DRM client ID to drm_sched_job_init

Adapt for upstream API change.

I'm really unsure if I missed a call site where an ID should be used
instead of 0.

Link: https://git.kernel.org/torvalds/c/2956554823ce
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 include/loonggpu_helper.h     | 7 +++++--
 include/loonggpu_job.h        | 2 +-
 loonggpu/loonggpu_cs.c        | 3 ++-
 loonggpu/loonggpu_job.c       | 4 ++--
 loonggpu/loonggpu_ttm.c       | 6 +++---
 loonggpu/loonggpu_ttm_buddy.c | 4 ++--
 loonggpu/loonggpu_vm.c        | 6 +++---
 7 files changed, 18 insertions(+), 14 deletions(-)

diff --git a/include/loonggpu_helper.h b/include/loonggpu_helper.h
index a761b3a..8f7715c 100644
--- a/include/loonggpu_helper.h
+++ b/include/loonggpu_helper.h
@@ -6,6 +6,7 @@
 #include <linux/mmu_context.h>
 #include <linux/kthread.h>
 #include <linux/cpufreq.h>
+#include <linux/version.h>
 #include <drm/gpu_scheduler.h>
 #include <drm/drm_syncobj.h>
 #include <drm/drm_cache.h>
@@ -1214,9 +1215,11 @@ static inline void lg_ring_sched_thread_unpark(struct loonggpu_ring *ring)
 
 static inline int lg_drm_sched_job_init(struct drm_sched_job *job,
 					struct drm_sched_entity *entity,
-					u32 credits, void *owner)
+					u32 credits, void *owner, u64 client_id)
 {
-#if defined(LG_DRM_SCHED_JOB_INIT_HAS_CREDITS)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 17, 0)
+	return drm_sched_job_init(job, entity, credits, owner, client_id);
+#elif defined(LG_DRM_SCHED_JOB_INIT_HAS_CREDITS)
 	return drm_sched_job_init(job, entity, credits, owner);
 #else
 	return drm_sched_job_init(job, entity, owner);
diff --git a/include/loonggpu_job.h b/include/loonggpu_job.h
index ca6b808..bf6d32c 100644
--- a/include/loonggpu_job.h
+++ b/include/loonggpu_job.h
@@ -43,7 +43,7 @@ int loonggpu_job_alloc_with_ib(struct loonggpu_device *adev, unsigned size,
 void loonggpu_job_free_resources(struct loonggpu_job *job);
 void loonggpu_job_free(struct loonggpu_job *job);
 int loonggpu_job_submit(struct loonggpu_job *job, struct drm_sched_entity *entity,
-		      void *owner, struct dma_fence **f);
+		      void *owner, struct dma_fence **f, u64 client_id);
 int loonggpu_job_submit_direct(struct loonggpu_job *job, struct loonggpu_ring *ring,
 			     struct dma_fence **fence);
 
diff --git a/loonggpu/loonggpu_cs.c b/loonggpu/loonggpu_cs.c
index 70c6dac..c3a7e48 100644
--- a/loonggpu/loonggpu_cs.c
+++ b/loonggpu/loonggpu_cs.c
@@ -1184,7 +1184,8 @@ static int loonggpu_cs_submit(struct loonggpu_cs_parser *p,
 	job = p->job;
 	p->job = NULL;
 
-	r = lg_drm_sched_job_init(&job->base, entity, 1, p->filp);
+	r = lg_drm_sched_job_init(&job->base, entity, 1, p->filp,
+				  p->filp->client_id);
 	if (r)
 		goto error_unlock;
 
diff --git a/loonggpu/loonggpu_job.c b/loonggpu/loonggpu_job.c
index 0e3114e..e3c03ed 100644
--- a/loonggpu/loonggpu_job.c
+++ b/loonggpu/loonggpu_job.c
@@ -118,7 +118,7 @@ void loonggpu_job_free(struct loonggpu_job *job)
 }
 
 int loonggpu_job_submit(struct loonggpu_job *job, struct drm_sched_entity *entity,
-		      void *owner, struct dma_fence **f)
+		      void *owner, struct dma_fence **f, u64 client_id)
 {
 	enum drm_sched_priority priority;
 	struct loonggpu_ring *ring;
@@ -127,7 +127,7 @@ int loonggpu_job_submit(struct loonggpu_job *job, struct drm_sched_entity *entit
 	if (!f)
 		return -EINVAL;
 
-	r = lg_drm_sched_job_init(&job->base, entity, 1, owner);
+	r = lg_drm_sched_job_init(&job->base, entity, 1, owner, client_id);
 	if (r)
 		return r;
 	lg_drm_sched_job_arm(&job->base);
diff --git a/loonggpu/loonggpu_ttm.c b/loonggpu/loonggpu_ttm.c
index 1745b98..2fd79b0 100644
--- a/loonggpu/loonggpu_ttm.c
+++ b/loonggpu/loonggpu_ttm.c
@@ -1971,7 +1971,7 @@ static int loonggpu_map_buffer(struct ttm_buffer_object *bo,
 		goto error_free;
 
 	r = loonggpu_job_submit(job, &adev->mman.entity,
-			      LOONGGPU_FENCE_OWNER_UNDEFINED, &fence);
+			      LOONGGPU_FENCE_OWNER_UNDEFINED, &fence, 0);
 	if (r)
 		goto error_free;
 
@@ -2043,7 +2043,7 @@ int loonggpu_copy_buffer(struct loonggpu_ring *ring, uint64_t src_offset,
 		r = loonggpu_job_submit_direct(job, ring, fence);
 	else
 		r = loonggpu_job_submit(job, &adev->mman.entity,
-				      LOONGGPU_FENCE_OWNER_UNDEFINED, fence);
+				      LOONGGPU_FENCE_OWNER_UNDEFINED, fence, 0);
 	if (r)
 		goto error_free;
 
@@ -2135,7 +2135,7 @@ int loonggpu_fill_buffer(struct loonggpu_bo *bo,
 	loonggpu_ring_pad_ib(ring, &job->ibs[0]);
 	WARN_ON(job->ibs[0].length_dw > num_dw);
 	r = loonggpu_job_submit(job, &adev->mman.entity,
-			      LOONGGPU_FENCE_OWNER_UNDEFINED, fence);
+			      LOONGGPU_FENCE_OWNER_UNDEFINED, fence, 0);
 	if (r)
 		goto error_free;
 
diff --git a/loonggpu/loonggpu_ttm_buddy.c b/loonggpu/loonggpu_ttm_buddy.c
index 050fbbb..7ce82af 100644
--- a/loonggpu/loonggpu_ttm_buddy.c
+++ b/loonggpu/loonggpu_ttm_buddy.c
@@ -206,7 +206,7 @@ static int loonggpu_ttm_map_buffer(struct ttm_buffer_object *bo,
 	}
 
 	r = loonggpu_job_submit(job, &adev->mman.entity,
-			     LOONGGPU_FENCE_OWNER_UNDEFINED, &fence);
+			     LOONGGPU_FENCE_OWNER_UNDEFINED, &fence, 0);
 	if (r)
 		loonggpu_job_free(job);
 
@@ -271,7 +271,7 @@ static int loonggpu_ttm_fill_mem(struct loonggpu_ring *ring, uint32_t src_data,
 	loonggpu_ring_pad_ib(ring, &job->ibs[0]);
 	WARN_ON(job->ibs[0].length_dw > num_dw);
 	r = loonggpu_job_submit(job, &adev->mman.entity,
-			     LOONGGPU_FENCE_OWNER_UNDEFINED, fence);
+			     LOONGGPU_FENCE_OWNER_UNDEFINED, fence, 0);
 	if (r)
 		loonggpu_job_free(job);
 
diff --git a/loonggpu/loonggpu_vm.c b/loonggpu/loonggpu_vm.c
index 1a6f3c9..7502441 100644
--- a/loonggpu/loonggpu_vm.c
+++ b/loonggpu/loonggpu_vm.c
@@ -360,7 +360,7 @@ static int loonggpu_vm_clear_bo(struct loonggpu_device *ldev,
 	if (r)
 		goto error_free;
 
-	r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_UNDEFINED, &fence);
+	r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_UNDEFINED, &fence, 0);
 	if (r)
 		goto error_free;
 
@@ -978,7 +978,7 @@ restart:
 		loonggpu_ring_pad_ib(ring, params.ib);
 		loonggpu_sync_resv(ldev, &job->sync, resv, LOONGGPU_SYNC_ALWAYS, LOONGGPU_FENCE_OWNER_VM, false);
 		WARN_ON(params.ib->length_dw > ndw);
-		r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_VM, &fence);
+		r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_VM, &fence, 0);
 		if (r)
 			goto error;
 
@@ -1235,7 +1235,7 @@ int loonggpu_vm_bo_update_mapping(struct loonggpu_device *ldev,
 
 	loonggpu_ring_pad_ib(ring, params.ib);
 	WARN_ON(params.ib->length_dw > ndw);
-	r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_VM, &f);
+	r = loonggpu_job_submit(job, &vm->entity, LOONGGPU_FENCE_OWNER_VM, &f, 0);
 	if (r)
 		goto error_free;
 
-- 
2.52.0

